File: boya_integration/public/js/budget.js
frappe.ui.form.on('Budget Account', {
    budget_amount: function(frm, cdt, cdn) {
        // When total budget in accounts is updated, recalculate total budget amount
        calculateTotalBudget(frm);
    }
});

function calculateTotalBudget(frm) {
    var total_budget = 0;
    frm.doc.accounts.forEach(function(row) {
        total_budget += (row.budget_amount || 0);
    });
    frm.set_value('total_budget', total_budget);
    frm.set_value('total_budget2', total_budget);
}

File: boya_integration/boya_integration/custom_methods/budget.py
import frappe

def before_save(doc, method=None):
    '''
    Extend method of the budget doctype that runs before the doctype is saved
    '''
    # Calculate the correct budget total
    correct_total_budget_amounts(doc)

def correct_total_budget_amounts(doc):
    total_budget = 0
    for account in doc.accounts:
        total_budget += account.budget_amount
    
    if total_budget != doc.total_budget2 or total_budget != doc.total_budget:
        doc.total_budget2 = total_budget
        doc.total_budget = total_budget
